pkg/catalogv2/git/download.go

```diff
@@ -64,11 +69,15 @@ func Ensure(secret *corev1.Secret, namespace, name, gitURL, commit string, insec
 		return err
 	}
 
+	if isBundled(git) && settings.SystemCatalog.Get() == "bundled" {
```

Comment:
@manno Correct me if I am wrong. Rancher updates the local repos by comparing the remote SHA and the clusterrepo.status.commit. If there is a difference it does git fetch and git reset to FETCH_HEAD here. https://github.com/rancher/rancher/blob/release/v2.7/pkg/catalogv2/git/git.go#L124

But this behavior doesn't happen in local catalogs in air-gapped mode. Link - https://github.com/rancher/rancher/blob/release/v2.7/pkg/catalogv2/git/download.go#L46